src/transformers/sagemaker/training_args_sm.py

```diff
@@ -24,33 +26,48 @@
 
 logger = logging.get_logger(__name__)
 
+# TODO: should be moved to `file_utils` after refactoring of SageMakerTrainer
 
-def is_smdistributed_available():
+
+def is_sagemaker_model_parallel_available():
+    # Get the sagemaker specific mp parameters from smp_options variable.
+    smp_options = os.getenv("SM_HP_MP_PARAMETERS", "{}")
+    try:
+        # Parse it and check the field "partitions" is included, it is required for model parallel.
+        smp_options = json.loads(smp_options)
+        if "partitions" not in smp_options:
+            return False
+    except json.JSONDecodeError:
+        return False
+
+    # Get the sagemaker specific framework parameters from mpi_options variable.
+    mpi_options = os.getenv("SM_FRAMEWORK_PARAMS", "{}")
+    try:
+        # Parse it and check the field "sagemaker_distributed_dataparallel_enabled".
+        mpi_options = json.loads(mpi_options)
+        if not mpi_options.get("sagemaker_mpi_enabled", False):
+            return False
+    except json.JSONDecodeError:
+        return False
+    # Lastly, check if the `smdistributed` module is present.
     return importlib.util.find_spec("smdistributed") is not None
 
 
-if is_smdistributed_available():
+if is_sagemaker_model_parallel_available():
     import smdistributed.modelparallel.torch as smp
 
+    smp.init()
+
 
 @dataclass
 class SageMakerTrainingArguments(TrainingArguments):
-    mp_parameters: str = field(
-        default="", metadata={"help": "Used by the SageMaker launcher to send mp-specific args."}
-    )
```

Comment:
Will this work with previous versions of `smdistributed` that were sending those parameters to the script?

Reply:
Added them again. SageMaker is passing `mp_parameters key=value,key=value` as args. https://github.com/huggingface/transformers/pull/10122/files#r574206511